src/include/platform/PersistedStorage.h

```diff
@@ -87,6 +91,139 @@ CHIP_ERROR Read(Key aKey, uint32_t & aValue);
  */
 CHIP_ERROR Write(Key aKey, uint32_t aValue);
 
+class KeyValueStore
+{
+public:
+    virtual ~KeyValueStore() = default;
+    /**
+     * @brief
+     *   Reads the value of an entry in the KVS. The value is read into the
+     *   provided buffer and the number of bytes read is returned. If desired,
+     *   the read can be started at an offset.
+     *
+     *   If the output buffer is too small for the value, Get returns
+     *   CHIP_ERROR_BUFFER_TOO_SMALL with the number of bytes read. The remainder of the
+     *   value can be read by calling get with an offset.
+     *
+     * @param[in]     key               The name of the key in which to get.
```

Comment:
```suggestion
     * @param[in]     key               The name of the key to get.
```

We're not going to worry about keys that might be integers or strings containing null or random byte arrays?

Reply:
Added a comment, should be a null terminated string.